texture.h

```diff
@@ -6,13 +6,20 @@ class Texture
 {
 private:
     std::string mFilePath;
-    unsigned char* mLocalBuffer;
-    int mWidth, mHeight, mBPP;
+    unsigned char* mLocalBuffer { nullptr };
 protected:
-    unsigned int mRendererID;
+    int mWidth {0};
+    int mHeight {0};
+    int mBPP {0};
+    unsigned int mRendererID {0};
 public:
+    Texture();
     Texture(const std::string& filePath);
```

Comment:
You should set this as "explicit" 

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c46-by-default-declare-single-argument-constructors-explicit

"C.46: By default, declare single-argument constructors explicit"